From: Spaceghost <Spaceghost@users.noreply.github.com>
Date: Sun, 30 Aug 2026 20:41:00 -0700
Subject: [PATCH] llseek: use syscall instead of obsolete _syscall5 macro

musl does not provide the legacy kernel _syscall5 macro.  Call syscall(2)
directly, as this source already does on i386.  This also avoids relying on
the implicit-int extension rejected by modern Clang/Zig.

diff --git a/lib/ext2fs/llseek.c b/lib/ext2fs/llseek.c
index 45f9a80..c1590db 100644
--- a/lib/ext2fs/llseek.c
+++ b/lib/ext2fs/llseek.c
@@ -63,15 +63,6 @@ extern long long llseek (int fd, long long offset, int origin);
 #define __NR__llseek            140
 #endif
 
-#ifndef __i386__
-static int _llseek (unsigned int, unsigned long,
-		   unsigned long, ext2_loff_t *, unsigned int);
-
-static _syscall5(int,_llseek,unsigned int,fd,unsigned long,offset_high,
-		 unsigned long, offset_low,ext2_loff_t *,result,
-		 unsigned int, origin)
-#endif
-
 static ext2_loff_t my_llseek (int fd, ext2_loff_t offset, int origin)
 {
 	ext2_loff_t result;
@@ -79,10 +70,6 @@ static ext2_loff_t my_llseek (int fd, ext2_loff_t offset, int origin)
 
-#ifndef __i386__
-	retval = _llseek(fd, ((unsigned long long) offset) >> 32,
-#else
 	retval = syscall(__NR__llseek, fd, (unsigned long long) (offset >> 32),
-#endif
-			  ((unsigned long long) offset) & 0xffffffff,
+			 ((unsigned long long) offset) & 0xffffffff,
 			&result, origin);
 	return (retval == -1 ? (ext2_loff_t) retval : result);
 }
